src/transaction/log_page_buffer.c

```diff
@@ -1554,19 +1558,108 @@ logpb_fetch_header_from_active_log (THREAD_ENTRY * thread_p, const char *db_full
   if (log_pgptr->hdr.logical_pageid != LOGPB_HEADER_PAGE_ID || log_pgptr->hdr.offset != NULL_OFFSET)
     {
       er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_PAGE_CORRUPTED, 1, phy_pageid);
+      // TODO: above error message says:
+      //  Internal error: logical log page %1$lld may be corrupted.
+      // but a phys page is is logged; should not LOGPB_HEADER_PAGE_ID actually be logged?
       error_code = ER_LOG_PAGE_CORRUPTED;
       goto error;
     }
 
+error:
+  return error_code;
+}
+
+#if defined (SERVER_MODE)
+/*
+ * logpb_fetch_active_log_header_from_page_server - Fetch log header from page server
+ *
+ * return: error code
+ *
+ *   hdr(in/out): Pointer where log header is to be copied
+ *   log_pgptr(in/out): log page buffer ptr
+ *
+ * NOTE: Should be used only: during boot sequence; on a transaction server with remote storage
+ */
+// *INDENT-OFF*
+int
+logpb_fetch_active_log_header_from_page_server (LOG_HEADER * hdr, LOG_PAGE * log_pgptr)
+{
+  assert (is_tran_server_with_remote_storage ());
+
+  request_log_page_from_ps (LOGPB_HEADER_PAGE_ID);
+
+  std::shared_ptr<log_page_owner> log_page_from_page_server
+    = ats_Gl.get_log_page_broker ().wait_for_page (LOGPB_HEADER_PAGE_ID);
+
+  const LOG_PAGE *const log_page_from_page_server_pgptr = log_page_from_page_server->get_log_page ();
```

Comment:
Suggestion for shorter names:
```suggestion
  std::shared_ptr<log_page_owner> brokered_log_hdr_page
    = ats_Gl.get_log_page_broker ().wait_for_page (LOGPB_HEADER_PAGE_ID);

  const LOG_PAGE *const log_hdr_page = log_page_from_page_server->get_log_page ();
```
The context set by the function name clarifies that the log page is from the page server. I don't think mentioning it in every variable is necessary. Very long names make the code actually harder to read.

Reply:
Shortened